CMakeLists.txt

```diff
@@ -23,27 +23,61 @@ endif()
 
 #do not add or build any tests of the dependencies
 set(original_run_e2e_tests ${run_e2e_tests})
+set(original_run_int_tests ${run_int_tests})
 set(original_run_unittests ${run_unittests})
+set(original_skip_samples ${skip_samples})
 
 set(run_e2e_tests OFF)
+set(run_int_tests OFF)
 set(run_unittests OFF)
+set(skip_samples ON)
 
-add_subdirectory(./deps/c-utility EXCLUDE_FROM_ALL)
+if(NOT ${use_installed_dependencies})
+    if (NOT TARGET azure_macro_utils_c)
```

Comment:
> if (NOT TARGET azure_macro_utils_c) [](start = 3, length = 36)

This should also check that the CMakeLists exists. The reason is that if you are going to do IoT-SDK-C without recursive, the deps/azure-macro-utils-c does not exist at all.

Same for the others. It is a good guard to have I believe.

Reply:
Seems like some of the others have it. Looks like macro utils and C shared need it.

---
In reply to: [271398582](https://github.com/Azure/azure-utpm-c/pull/23#discussion_r271398582) [](ancestors = 271398582)